lxd/instance/drivers/driver_lxc.go

```diff
@@ -632,6 +632,12 @@ func (d *lxc) initLXC(config bool) error {
 		return err
 	}
 
+	// To discuss: take defalt timeout value from some global config?
+	err = liblxc.SetTimeout(time.Duration(10) * time.Second)
```

Comment:
I don't think we need to make it configurable.
However what happens if we, say, set a 10s timeout by default on init, as you're doing here, will this affect a longer timeout passed to d.c.Shutdown()?

Reply:
Also, can se set the timeout on the `cc` returned from `liblxc.NewContainer` rather than on the library as a whole?

Reply:
> Also, can se set the timeout on the `cc` returned from `liblxc.NewContainer` rather than on the library as a whole?

sure, it's just mistake. Thanks!

Reply:
> I don't think we need to make it configurable. However what happens if we, say, set a 10s timeout by default on init, as you're doing here, will this affect a longer timeout passed to d.c.Shutdown()?

Yep, that's what I've said in the original issue. That's why right now this timeout is not applied to all commands. We will need to extend liblxc and each command one by one to support this timeout. [like this https://github.com/lxc/lxc/pull/4260/commits/47630db7970022d7b174bbc631db4cd86f6a8fcb]And I'm not sure that we really need that. So, it's thing to discuss.

Reply:
OK so right now this would just record the timeout value but not use it, and then you would extend each underlying call to use it. And we wouldn't need to update d.c.Shutdown() to use it because it already has its own timeout support?

Reply:
> OK so right now this would just record the timeout value but not use it, and then you would extend each underlying call to use it. And we wouldn't need to update d.c.Shutdown() to use it because it already has its own timeout support?

sure, let's do it. I've started from state command because it was the original problem. I'll add Shutdown support too. ;-)

Reply:
No, you misunderstand me. I don't think we need to add support to `Shutdown` because it already has its own timeout argument.

I was asking whether it needed to take into account the container-wide timeout value because it already has its own one.
Basically what ever is passed to `Shutdown` should take precedent over the timeout applied to the container IMHO.

Reply:
Right now, for `Shutdown` this container-wide timeout will be ignored at all.

You want to propose to take this "container-wide timeout" as a default value for `Shutdown` call (if timeout specified as `-1`)?

Reply:
well in Go at least we cannot specify -1, but we could have it use the container wide timeout if the timeout passed was zero perhaps?

Reply:
That's an interesting point.

What we have in the `liblxc`:
```
	/*!
	 * \brief Request the container shutdown by sending it \c
	 * SIGPWR.
	 *
	 * \param c Container.
	 * \param timeout Seconds to wait before returning false.
	 *  (-1 to wait forever, 0 to avoid waiting).
	 *
	 * \return \c true if the container was shutdown successfully, else \c false.
	 */
	bool (*shutdown)(struct lxc_container *c, int timeout);
```

In the `go-lxc`:
```
// Shutdown shuts down the container.
func (c *Container) Shutdown(timeout time.Duration) error {
	c.mu.Lock()
	defer c.mu.Unlock()

	if c.container == nil {
		return ErrNotDefined
	}

	if err := c.makeSure(isRunning); err != nil {
		return err
	}

	if !bool(C.go_lxc_shutdown(c.container, C.int(timeout.Seconds()))) {
		return ErrShutdownFailed
	}
	return nil
}
```

I don't think that it would be correct to change semantics of `liblxc` `shutdown` call. But we can change `go-lxc`'s `Shutdown` implementation to take the "container-wide timeout" as a default in case when `timeout` is 0. Or, better, to add another method like `ShutdownWithTimeout`.

Reply:
We don't need `ShutdownWithTimeout` as we already have `Shutdown(timeout time.Duration)`.

I propose we just ignore the container wide timeout for the `Shutdown(timeout time.Duration)` function then.

Reply:
> I propose we just ignore the container wide timeout for the `Shutdown(timeout time.Duration)` function then.

It's just the current implementation behavior.